testsuite/smoke-test/src/state_sync.rs

```diff
@@ -219,82 +214,51 @@ async fn test_full_node_continuous_sync_transactions() {
-    test_full_node_sync(vfn_peer_id, &mut swarm, false, false).await;
-}
-
-/// Creates a new full node using the given config and swarm
-async fn create_full_node(full_node_config: NodeConfig, swarm: &mut LocalSwarm) -> PeerId {
-    let validator_peer_id = swarm.validators().next().unwrap().peer_id();
-    let vfn_peer_id = swarm
-        .add_validator_fullnode(
-            &swarm.versions().max().unwrap(),
-            OverrideNodeConfig::new_with_default_base(full_node_config),
-            validator_peer_id,
-        )
-        .unwrap();
-    for fullnode in swarm.full_nodes_mut() {
-        fullnode
-            .wait_until_healthy(Instant::now() + Duration::from_secs(MAX_HEALTHY_WAIT_SECS))
-            .await
-            .unwrap();
-    }
-    vfn_peer_id
+    // Create the fullnode and test its ability to sync
+    let vfn_peer_id = create_fullnode(vfn_config, &mut swarm).await;
+    test_fullnode_sync(vfn_peer_id, &mut swarm, false, false).await;
 }
 
-/// A helper method that tests that a full node can sync from a validator after
-/// a failure and continue to stay up-to-date.
-async fn test_full_node_sync(
-    vfn_peer_id: PeerId,
-    swarm: &mut LocalSwarm,
-    epoch_changes: bool,
-    clear_storage: bool,
-) {
-    // Stop the fullnode and potentially clear storage
-    if clear_storage {
-        stop_fullnode_and_delete_storage(swarm, vfn_peer_id).await;
-    } else {
-        swarm.fullnode_mut(vfn_peer_id).unwrap().stop();
-    }
+#[tokio::test]
+async fn test_single_validator_reboot() {
+    // Create a swarm of 1 validator
+    let mut swarm = new_local_swarm_with_aptos(1).await;
 
-    // Execute a number of transactions on the validator
-    let validator_peer_id = swarm.validators().next().unwrap().peer_id();
-    let validator_client = swarm.validator(validator_peer_id).unwrap().rest_client();
-    let (mut account_0, mut account_1) = create_test_accounts(swarm).await;
+    // Execute multiple transactions
+    let validator = swarm.validators_mut().next().unwrap();
+    let validator_client = validator.rest_client();
+    let (mut account_0, mut account_1) = create_test_accounts(&mut swarm).await;
     execute_transactions(
-        swarm,
+        &mut swarm,
         &validator_client,
         &mut account_0,
         &account_1,
-        epoch_changes,
+        true,
     )
     .await;
 
-    // Restart the fullnode and verify it can sync
+    // Restart the validator
+    let validator = swarm.validators_mut().next().unwrap();
+    validator.stop();
+    validator.start().unwrap();
     swarm
-        .fullnode_mut(vfn_peer_id)
-        .unwrap()
-        .restart()
+        .wait_all_alive(Duration::from_secs(MAX_CATCH_UP_WAIT_SECS))
         .await
         .unwrap();
-    wait_for_all_nodes(swarm).await;
 
-    // Execute more transactions on the validator and verify the fullnode catches up
-    execute_transactions_and_wait(
-        swarm,
+    // Execute more transactions
+    execute_transactions(
```

Comment:
Aah, this is because of the diff 😄 This didn't change (AFAICT). But, it's a good question. We should be able to wait. I'll update it in this test. The current version is: https://github.com/aptos-labs/aptos-core/blob/main/testsuite/smoke-test/src/state_sync.rs#L858

Reply:
Ahhh, we need AI to do more semantic-aware diffs